### ISSUE TITLE
Traffic not routed through WAN

### ISSUE THREAD

**Issue by machinist22** [Ranger802004/asusmerlin#59]
Hey there. Trying to get this to work for a local streaming service, that won't allow foreign traffic. The device is hooked up via VPN. Here's what I've done after installing:

1. Created policy:

- given it a name

- selected "wan" as interface

- y to verbose logging

- no to private IP's

- y to CNAMES

2. Have gathered all the hostnames and have opted for adding them in bulk:

Added them to `/jffs/configs/domain_vpn_routing/policy_Example_domainlist` (making sure to leave a line empty), saved, exited.

3. Ran `/jffs/scripts/domain_vpn_routing.sh querypolicy all` -- it went through each of the hostnames, which all in all took around 5 minutes. Then printed:

***New IP Addresses detected for TV***
Updating Policy: TV
Updated Policy: TV
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 156.231.218.12 to IPSET: DVR-TV-v4
domain_vpn_routing: Query Policy - Processing Time: 894 seconds

4. Set the boot delay to 30 seconds to allow the VPN connections to start.

5. Rebooted 

When I start up the app, I'm still greeted with the message of being from outside the country.

Questions:

Do I need to enable Private IP's even when I already have rules for this in VPN Director, as in "only route this IP through OVPN1"?

I have Skynet installed - would that interfere somehow? Also using VPNMON-R3.

What else can I try?

Any pointers greatly appreciated.

**Comment by Ranger802004** [comment]
It looks like there is an issue with the length of the comment being added in the ipset for the domain(s), what domains are you trying to route?

**Comment by machinist22** [comment]
I retried it all to test with the Analiti speedtest app that's available on Android TV.

I gathered all hostnames:

> checkip.amazonaws.com
firebaselogging.googleapis.com
ookla1.speedtest.tng.de
play-fe.googleapis.com
speedtest-iks.altibox.net
speedtest.ansluten.net
speedtest.mlm.telenor.se
www.google.com
www.speedtest.net

I then queried them, which worked without errors. I then rebooted the router AND the device, as well as cleared data on the app first.

It still tells me I'm using the VPN connection.

Can you tell me what I'm doing wrong or what I should try exactly?

**Comment by Ranger802004** [comment]
> I retried it all to test with the Analiti speedtest app that's available on Android TV.
> 
> I gathered all hostnames:
> 
> > checkip.amazonaws.com
> > firebaselogging.googleapis.com
> > ookla1.speedtest.tng.de
> > play-fe.googleapis.com
> > speedtest-iks.altibox.net
> > speedtest.ansluten.net
> > speedtest.mlm.telenor.se
> > [www.google.com](http://www.google.com)
> > [www.speedtest.net](http://www.speedtest.net)
> 
> I then queried them, which worked without errors. I then rebooted the router AND the device, as well as cleared data on the app first.
> 
> It still tells me I'm using the VPN connection.
> 
> Can you tell me what I'm doing wrong or what I should try exactly?

When you reboot you will have to allow the Domain VPN Routing to run and restore policies, nothing is saved in the environment and is loaded into memory so it has to be restored after a reboot.

**Comment by machinist22** [comment]
> When you reboot you will have to allow the Domain VPN Routing to run and restore policies, nothing is saved in the environment and is loaded into memory so it has to be restored after a reboot.

Got it. Having retried without rebooting, I am still identified as a VPN user. It does not appear that the app is using WAN.

Can you tell me what I'm doing wrong or what I should try exactly?

Am I correct in saying `yes` to both `Private IP's` and `CNAMES`?

Thank you

**Comment by Ranger802004** [comment]
Verify you have all of the appropriate domains? I recommend to use the IPFoo browser extension to do this.  Also allow time for the query policies to run and collect IPs if it can query different addresses periodically.  For a new policy let it sit and run for awhile to continue querying the domains continuously.  Once the IPs are all queried they are stored into the policy for restoration of the policy during reboot/etc.

**Comment by machinist22** [comment]
> Verify you have all of the appropriate domains?

I would think so. I use DNS Masq logging via https://github.com/Xentrk/x3mRouting#getdomainnamessh-script directly on the device when I'm navigating the streaming app.

Question:

I have a bunch of akamai hostnames, like:

srv1.akamaized.net
srv1.akamaized.net
srv1.akamaized.net

**Comment by machinist22** [comment]
> It looks like there is an issue with the length of the comment being added in the ipset for the domain(s), what domains are you trying to route?

I'm also still getting the error:

`ipset v7.6: Syntax error: Comment is longer than the maximum allowed 255 characters
domain_vpn_routing: Query Policy - ***Error*** Failed to add 18.173.5.109 to IPSET: DVR-TV-v4`

But I don't control "comments" here- how can I?

**Comment by Ranger802004** [comment]
The comments are the domain names for the particular IP address in the IPSet, if multiple domains query to the same address they are added in a comma delimited comment.  I'll have to make an update that checks for the length and come up with another way to accommodate that.

**Comment by machinist22** [comment]
> The comments are the domain names for the particular IP address in the IPSet, if multiple domains query to the same address they are added in a comma delimited comment. I'll have to make an update that checks for the length and come up with another way to accommodate that.

That would terrific

**Comment by Ranger802004** [comment]
> > The comments are the domain names for the particular IP address in the IPSet, if multiple domains query to the same address they are added in a comma delimited comment. I'll have to make an update that checks for the length and come up with another way to accommodate that.
> 
> 
> 
> That would be terrific

I will try to work this into the next patch.

**Comment by machinist22** [comment]
Thank you, looking forward to it

**Comment by Ranger802004** [comment]
v3.0.5-beta1 has been published to the beta channel (Enable Dev Mode to update to beta versions).

Release Notes:
v3.0.5-beta1 - 12/30/2024
Fixes:
- Fixed an issue where comments for IPSets were greater than 255 and causing the IP Addresses to not be added.
- Fixed an issue for errors being generated from null data being returned from AdGuardHome log parsing and ASN queries.
- Fixed other minor issues

**Comment by machinist22** [comment]
Damn, that was fast. I'll give it a try, and thank you

**Comment by machinist22** [comment]
It worked, no errors this time.

Question, because it's not clear from the readme:

1.  After having queried the policy and it has run, then what? Do I need to reboot the router, or is it ready as is?

2. _Can_ I reboot? If so, anything I need to do to have the job running again?

**Comment by Ranger802004** [comment]
> It worked, no errors this time.
> 
> 
> 
> Question, because it's not clear from the readme:
> 
> 
> 
> 1.  After having queried the policy and it has run, then what? Do I need to reboot the router, or is it ready as is?
> 
> 
> 
> 2. _Can_ I reboot? If so, anything I need to do to have the job running again?

You do not need to reboot, it will take immediate effect.  If you do reboot there is a job that will restore existing policies using the restorepolicy function.

**Comment by machinist22** [comment]
> > It worked, no errors this time.
> > Question, because it's not clear from the readme:
> > 
> > 1. After having queried the policy and it has run, then what? Do I need to reboot the router, or is it ready as is?
> > 2. _Can_ I reboot? If so, anything I need to do to have the job running again?
> 
> You do not need to reboot, it will take immediate effect. If you do reboot there is a job that will restore existing policies using the restorepolicy function.

Yeah, I see it. So if I reboot the router, I'll have to also "restore" here?

**Comment by Ranger802004** [comment]
> > > It worked, no errors this time.
> 
> > > Question, because it's not clear from the readme:
> 
> > > 
> 
> > > 1. After having queried the policy and it has run, then what? Do I need to reboot the router, or is it ready as is?
> 
> > > 2. _Can_ I reboot? If so, anything I need to do to have the job running again?
> 
> > 
> 
> > You do not need to reboot, it will take immediate effect. If you do reboot there is a job that will restore existing policies using the restorepolicy function.
> 
> 
> 
> Yeah, I see it. So if I reboot the router, I'll have to also "restore" here?
> 
> 

No it will execute at start up using wan-event or one of the vpn event scripts.

**Comment by machinist22** [comment]
Help me understand the wildcard: If I have a bunch of hostnames like 

`drod20a.akamaized.net
drod20m.akamaized.net
drod20o.akamaized.net
drod21c.akamaized.net
drod21f.akamaized.net
drod21n.akamaized.net
drod22g.akamaized.net
drod22k.akamaized.net
drod22l.akamaized.net
drod24b.akamaized.net
drod24f.akamaized.net
drod24o.akamaized.net
drod24r.akamaized.net`

-would I be best served just adding the `*.akamaized.net` domain to the policy?

**Comment by Ranger802004** [comment]
Wildcards are captured passively from the dnsmasq log, I would recommend to add the domains if you have them already identified.

**Comment by Ranger802004** [comment]
Following up with you, any updates?

**Comment by machinist22** [comment]
Thanks @Ranger802004 -- I've put in all the hostnames I can possibly find, both via DNS Masq and via the browser extension IPFoo. I've then entered `*.akamaized.net` just to be sure. My problem is it is still not really working - I'm still being detected as being out of country.

My thought was to just keep it running and perhaps it would catch any new hostnames from the wildcards? Idk, but so far nothing.

**Comment by machinist22** [comment]
I did already in the forum, if you don't mind

**Comment by Ranger802004** [comment]
> I did already in the forum, if you don't mind

I need them posted here please for reference tracking later, thank you.

**Comment by machinist22** [comment]
`64r04un2x4gxtacm0iybtrp8sbz3a1735995651.uaid.nmrodam.com
7ssw04dvtkwjqfnc1xa1db9l2xym91735995997.uaid.nmrodam.com
android.googleapis.com
androidtvchannels-pa.googleapis.com
app-measurement.com
cdn-gl.imrworldwide.com
clients3.google.com
cm.everesttech.net
connectivitycheck.gstatic.com
d3hya20wdoitcsqg4knxhutbigltp1735995883.uaid.nmrodam.com
downol.dr.dk
dpm.demdex.net
dr-massive.com
dr.d3.sc.omtrdc.net
dr.demdex.net
drevent01hls.akamaized.net
drlive01hls.akamaized.net
drlive02hls.akamaized.net
drlive03hls.akamaized.net
drod20c.akamaized.net
drod20h.akamaized.net
drod20j.akamaized.net
drod20n.akamaized.net
drod20o.akamaized.net
drod20s.akamaized.net
drod20t.akamaized.net
drod21a.akamaized.net
drod21h.akamaized.net
drod21k.akamaized.net
drod21l.akamaized.net
drod21n.akamaized.net
drod21p.akamaized.net
drod21q.akamaized.net
drod21t.akamaized.net
drod22c.akamaized.net
drod22f.akamaized.net
drod22g.akamaized.net
drod22i.akamaized.net
drod22m.akamaized.net
drod22n.akamaized.net
drod22s.akamaized.net
drod23a.akamaized.net
drod23b.akamaized.net
drod23d.akamaized.net
drod23i.akamaized.net
drod23k.akamaized.net
drod24e.akamaized.net
drod24f.akamaized.net
drod24j.akamaized.net
drod24k.akamaized.net
drod24n.akamaized.net
drod24o.akamaized.net
drod24r.akamaized.net
embeddedassistant.googleapis.com
firebase-settings.crashlytics.com
firebaseinstallations.googleapis.com
firebaselogging-pa.googleapis.com
firebaseremoteconfig.googleapis.com
klyvsgset60cvizrnle4o03s9k57e1735995933.uaid.nmrodam.com
login.dr.dk
mtalk.google.com
nexus.ensighten.com
play-fe.googleapis.com
play.googleapis.com
pro.ip-api.com
prod95-cdn.dr-massive.com
prod95-static.dr-massive.com
prod95.dr-massive.com
secure-dmk.nmrodam.com
stun.szbboys.com
tms.szbboys.com
update.googleapis.com
www.dr.dk
www.google.com
www.googleapis.com
www.youtube.com
www14.smartadserver.com`

**Comment by machinist22** [comment]
Following up with you here. Sorry, not sure if you were going to double check the hostnames

**Comment by Ranger802004** [comment]
Sorry I have been tied up, I will investigate soon.

**Comment by machinist22** [comment]
@Ranger802004 Was looking through the setup: Do I need to enable `WAN Reverse Path Filter` and if so, how would it help?

**Comment by Ranger802004** [comment]
> [@Ranger802004](https://github.com/Ranger802004) Was looking through the setup: Do I need to enable `WAN Reverse Path Filter` and if so, how would it help?

Do you have Dual WAN?

**Comment by machinist22** [comment]
No

**Comment by machinist22** [comment]
> Sorry I have been tied up, I will investigate soon.

Hi mate, so just following up here. Do you think you'll be able to assist with what's going on? If not, totally fine, just checking

**Comment by Ranger802004** [comment]
I have been away due to a family crisis, I will be picking back up on issues soon.  Apologies for the delay.

**Comment by Ranger802004** [comment]
Apologies for the delays, I am currently testing your policy now.

**Comment by Ranger802004** [comment]
What is the main URL for the service you are trying to route over the VPN with this policy?

**Comment by machinist22** [comment]
dr.dk

**Comment by Ranger802004** [comment]
I see you added www.dr.dk to your policy, did you add dr.dk without the www. subdomain?

**Comment by machinist22** [comment]
Yes, tried that as well. Including *.dr.dk

**Comment by Ranger802004** [comment]
I just added dr.dk to a test policy on top of the domains you gave me and it appears to be working.

**Comment by Ranger802004** [comment]
Try adding these additional domains to your policy.

> api.dr.dk
> asset.dr.dk

**Comment by machinist22** [comment]
Will do.

Question: I see `*.uaid.nmrodam.com` (asterisk being a new ID every time I check for hostnames) -- almost looks like they assign an ID to the session? How would I account for this -- is it correct to put the asterisk here?

**Comment by Ranger802004** [comment]
You can add the wildcard domain but it's only going to get picked up passively when the logs are reviewed and not active while you are browsing the site.  You could query all of those domains and see if they go to a common set of IP Addresses and route those IP Addresses possibly?

**Comment by Ranger802004** [comment]
Following up on your issues, any progress?

**Comment by machinist22** [comment]
> Following up on your issues, any progress?

Yeah thanks, it's not working. I've tried everything mentioned in here. I've reinstalled, manually followed the ID-type subdomains above and put in their hostnames (turns out is cloudfront), but I'm still detected as being outside the country.

**Comment by Ranger802004** [comment]
Start a new issue with the kill process issue in GitHub please.  Sounds like you might have to route some ASNs over your VPN possibly like Cloudfront in order for it to work.
